pkg/ovsdb/ovsdb.go

```diff
@@ -0,0 +1,329 @@
+	mutateOp := libovsdb.Operation{
+		Op:        "mutate",
+		Table:     "Bridge",
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+
+	// Perform OVS transaction
+	operations := []libovsdb.Operation{intfOp, portOp, mutateOp}
+	return self.ovsdbTransact(operations)
+}
+
+// Check if the bridge entry already exists
+func (self *OvsBridgeDriver) IsBridgePresent(bridgeName string) (bool, error) {
+	// lock the cache for read
+	self.lock.RLock()
+	defer self.lock.RUnlock()
+
+	condition := libovsdb.NewCondition("name", "==", bridgeName)
+	selectOp := libovsdb.Operation{
+		Op:      "select",
+		Table:   "Bridge",
+		Where:   []interface{}{condition},
+		Columns: []string{"name"},
+	}
+
+	transactionResult, err := self.ovsClient.Transact("Open_vSwitch", selectOp)
+	if err != nil {
+		return false, err
+	}
+
+	if len(transactionResult) != 1 {
+		return false, fmt.Errorf("unknow error")
+	}
+
+	operationResult := transactionResult[0]
+	if operationResult.Error != "" {
+		return false, fmt.Errorf("%s - %s", operationResult.Error, operationResult.Details)
+	}
+
+	if len(operationResult.Rows) != 1 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// Return ovs port name for an container interface
+func (self *OvsBridgeDriver) GetOvsPortForContIface(contIface, contNetnsPath string) (string, bool, error) {
+	searchMap := map[string]string{"contNetns": contNetnsPath, "contIface": contIface}
+	ovsmap, err := libovsdb.NewOvsMap(searchMap)
+	if err != nil {
+		return "", false, err
+	}
+
+	condition := libovsdb.NewCondition("external_ids", "==", ovsmap)
+	selectOp := libovsdb.Operation{
+		Op:      "select",
+		Table:   "Port",
+		Where:   []interface{}{condition},
+		Columns: []string{"name", "external_ids"},
+	}
+
+	transactionResult, err := self.ovsClient.Transact("Open_vSwitch", selectOp)
+	if err != nil {
+		return "", false, err
+	}
+
+	if len(transactionResult) != 1 {
+		return "", false, fmt.Errorf("unknow error")
+	}
+
+	operationResult := transactionResult[0]
+	if operationResult.Error != "" {
+		return "", false, fmt.Errorf("%s - %s", operationResult.Error, operationResult.Details)
+	}
+
+	if len(operationResult.Rows) != 1 {
+		return "", false, fmt.Errorf("error get more than one port")
+	}
+
+	return fmt.Sprintf("%v", operationResult.Rows[0]["name"]), true, nil
+}
+
+// ************************ Notification handler for OVS DB changes ****************
+func (self *OvsBridgeDriver) Update(context interface{}, tableUpdates libovsdb.TableUpdates) {
+}
+func (self *OvsBridgeDriver) Disconnected(ovsClient *libovsdb.OvsdbClient) {
+}
+func (self *OvsBridgeDriver) Locked([]interface{}) {
+}
+func (self *OvsBridgeDriver) Stolen([]interface{}) {
+}
+func (self *OvsBridgeDriver) Echo([]interface{}) {
+}
+
+// ************************ Helper functions ********************
+func createInterfaceOperation(intfName, ovnPortName string) ([]libovsdb.UUID, *libovsdb.Operation, error) {
```

Comment:
can we make the logic of this module simpler if we always use the same name for port and interface?

Reply:
the OVN port name is different from the veth name.